optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/ConstraintCollectors.java

```diff
@@ -1225,13 +1331,18 @@ private static Runnable innerCountDistinctLong(CountDistinctLongResultContainer
      */
     public static <A, B, C, D, Mapped extends Comparable<Mapped>> QuadConstraintCollector<A, B, C, D, ?, SortedSet<Mapped>>
             toSortedSet(QuadFunction<A, B, C, D, Mapped> groupValueMapping) {
```

Comment:
Separate issue: as we support toSortedSet(mapping), we should probably also support toSortedSet(mapping, Comparator). Not having it is an API gap.

Reply:
Making toMap() incremental in this way is likely impossible.
The interface of the method requires post-processing on the internal map before the collector returns.

Reply:
https://issues.redhat.com/browse/PLANNER-2405